test/functional/services/find.js

```diff
@@ -33,14 +34,31 @@ export function FindProvider({ getService }) {
 
     async existsByLinkText(linkText) {
       log.debug(`existsByLinkText ${linkText}`);
+      const remoteWithTimeout = remote.setFindTimeout(1000);
+      const exists = await remoteWithTimeout.findDisplayedByLinkText(linkText)
+        .then(() => true)
+        .catch(() => false);
+      remoteWithTimeout.setFindTimeout(defaultFindTimeout);
+      return exists;
+    }
 
-      const exists = await remote
-        .setFindTimeout(1000)
-        .findDisplayedByLinkText(linkText)
+    async existsByDisplayedByCssSelector(selector) {
+      log.debug(`existsByDisplayedByCssSelector ${selector}`);
+      const remoteWithTimeout = remote.setFindTimeout(1000);
+      const exists = await remoteWithTimeout.findDisplayedByCssSelector(selector)
         .then(() => true)
         .catch(() => false);
+      remoteWithTimeout.setFindTimeout(defaultFindTimeout);
+      return exists;
+    }
 
-      remote.setFindTimeout(defaultFindTimeout);
+    async existsByCssSelector(selector) {
+      log.debug(`existsByCssSelector ${selector}`);
+      const remoteWithTimeout = remote.setFindTimeout(1000);
+      const exists = await remoteWithTimeout.findByCssSelector(selector)
```

Comment:
Not a fan of mixing `async/await` with `.then().catch()`. A `try/catch` would probably pair nicely with `this.withTimeout()`